src/transformers/models/groupvit/modeling_groupvit.py

```diff
@@ -0,0 +1,1864 @@
+    """
+
+    attn_maps = []
+    with torch.no_grad():
+        prev_attn_masks = None
+        for attn_masks in attentions:
+            # [batch_size, num_groups, height x width] -> [batch_size, height x width, num_groups]
+            attn_masks = attn_masks.permute(0, 2, 1).contiguous()
+            if prev_attn_masks is None:
+                prev_attn_masks = attn_masks
+            else:
+                prev_attn_masks = prev_attn_masks @ attn_masks
+            # [B, HxW, G] -> [B, G, HxW] -> [B, G, H, W]
+            cur_attn_map = resize_attn_map(prev_attn_masks.permute(0, 2, 1).contiguous(), *hw_shape)
+            attn_maps.append(cur_attn_map)
+
+    # [B, G, H, W]
+    final_grouping = attn_maps[-1]
+
+    return final_grouping
+
+
+class GroupViTAttention(nn.Module):
+    def __init__(self, config: GroupViTVisionConfig):
+        super().__init__()
+        self.num_heads = config.num_attention_heads
+        head_dim = config.hidden_size // self.num_heads
+        self.scale = head_dim**-0.5
+
+        self.q_proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.k_proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.v_proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.proj = nn.Linear(config.hidden_size, config.hidden_size)
+
+    def forward(self, query, key=None):
+        batch_size, query_length, channels = query.shape
+        if key is None:
+            key = query
+        key_length = key.size(1)
+
+        # [batch_size, num_heads, query_length, channels//num_heads]
+        q = (
+            self.q_proj(query)
+            .reshape(batch_size, query_length, self.num_heads, channels // self.num_heads)
+            .permute(0, 2, 1, 3)
+        )
+        # [batch_size, num_heads, key_length, channels//num_heads]
+        k = (
+            self.k_proj(key)
+            .reshape(batch_size, key_length, self.num_heads, channels // self.num_heads)
+            .permute(0, 2, 1, 3)
+        )
+        # [batch_size, num_heads, key_length, channels//num_heads]
+        v = (
+            self.v_proj(key)
+            .reshape(batch_size, key_length, self.num_heads, channels // self.num_heads)
+            .permute(0, 2, 1, 3)
+        )
+
+        # [batch_size, num_heads, query_length, key_length]
+        attn = (q @ k.transpose(-2, -1)) * self.scale
+        attn = attn.softmax(dim=-1)
+        assert attn.shape == (
+            batch_size,
+            self.num_heads,
+            query_length,
+            key_length,
+        ), f"{attn.shape} vs {(batch_size, self.num_heads, query_length, key_length)}"
+
+        # [batch_size, num_heads, query_length, channels//num_heads] -> [batch_size, query_length, channels]
+        out = (attn @ v).transpose(1, 2).reshape(batch_size, query_length, channels)
+        out = self.proj(out)
+        return out
+
+
+class GroupViTCrossAttentionBlock(nn.Module):
```

Comment:
```suggestion
class GroupViTCrossAttentionLayer(nn.Module):
```
(nit) We typically use the term layer instead of block.